diff -Naur manokwari-1.0.9.orig/src/main.vala manokwari-1.0.9/src/main.vala
--- manokwari-1.0.9.orig/src/main.vala	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/src/main.vala	2016-08-24 15:46:29.786075251 +0700
@@ -19,14 +19,14 @@
     Intl.bind_textdomain_codeset( Config.GETTEXT_PACKAGE, "UTF-8" );
     Intl.textdomain( Config.GETTEXT_PACKAGE );
 
- 
+
     Gtk.init (ref args);
 
     var id = GLib.Environment.get_variable("DESKTOP_AUTOSTART_ID");
     var app = new Unique.App ("id.or.blankonlinux.Manokwari", id);
     if (app.is_running ()) {
         stdout.printf ("Manokwari is already running.\n");
-        return 0;    
+        return 0;
     }
 
     PanelSessionManager.getInstance ();
@@ -48,7 +48,7 @@
     var d = new PanelDesktop ();
     d.show ();
 
-    // Window 
+    // Window
     var w = new PanelWindowHost ();
     w.show();
 
@@ -62,7 +62,7 @@
             menu_box.show ();
         }
     });
-
+    
     w.dialog_opened.connect (() => {
         if (menu_box.visible) {
             menu_box.try_hide ();
@@ -81,6 +81,36 @@
         }
     });
 
+  //////////// extra ////////////
+    var extra_box = new PanelExtraBox();
+    // SIGNALS
+    w.extra_clicked.connect (() => {
+        if (extra_box.visible) {
+            extra_box.try_hide ();
+        } else {
+            // Otherwise we want to show it
+            extra_box.show ();
+        }
+    });
+
+    w.dialog_opened.connect (() => {
+        if (extra_box.visible) {
+            extra_box.try_hide ();
+        }
+    });
+
+    d.desktop_clicked.connect (() => {
+        if (extra_box.visible) {
+            extra_box.try_hide ();
+        }
+    });
+
+    w.windows_visible.connect (() => {
+        if (extra_box.visible) {
+            extra_box.try_hide ();
+        }
+    });
+
 
     Gtk.main ();
     return 0;
diff -Naur manokwari-1.0.9.orig/src/Makefile.am manokwari-1.0.9/src/Makefile.am
--- manokwari-1.0.9.orig/src/Makefile.am	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/src/Makefile.am	2016-08-17 01:52:46.000000000 +0700
@@ -1,4 +1,4 @@
-NULL = 
+NULL =
 
 AM_CPPFLAGS = \
 	$(WNCK_GMENU_DISCLAIMER) \
@@ -36,6 +36,9 @@
 		panel-shell.vala \
 		panel-socket.vala \
 		panel-xdg-properties.vala \
+		panel-extra-content.vala \
+		panel-extra-box.vala \
+		panel-extra-html.vala \
 	$(NULL)
 
 manokwari_VALAFLAGS = \
@@ -53,5 +56,3 @@
 
 DISTCLEANFILES = \
 	$(NULL)
-
-
diff -Naur manokwari-1.0.9.orig/src/panel-abstract-window.vala manokwari-1.0.9/src/panel-abstract-window.vala
--- manokwari-1.0.9.orig/src/panel-abstract-window.vala	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/src/panel-abstract-window.vala	2016-08-17 13:07:43.397575316 +0700
@@ -7,7 +7,7 @@
     protected void set_struts () {
         ulong[] struts = new ulong[12];
 
-        var r = PanelScreen.get_primary_monitor_geometry (); 
+        var r = PanelScreen.get_primary_monitor_geometry ();
 
         var left = 0;
         var right = 0;
@@ -89,7 +89,7 @@
         set_focus_on_map (true);
         set_accept_focus (true);
         stick ();
-     
+
         var screen = get_screen();
         screen.monitors_changed.connect (() => {
             screen_size_changed ();
diff -Naur manokwari-1.0.9.orig/src/panel-extra-box.vala manokwari-1.0.9/src/panel-extra-box.vala
--- manokwari-1.0.9.orig/src/panel-extra-box.vala	1970-01-01 07:00:00.000000000 +0700
+++ manokwari-1.0.9/src/panel-extra-box.vala	2016-08-24 18:02:15.415880035 +0700
@@ -0,0 +1,127 @@
+using Gtk;
+
+public class PanelExtraBox : PanelAbstractWindow {
+    private const int COLUMN_WIDTH = 265;
+    private const int TOP = 24;
+    private bool inside;
+
+    public signal void dismissed ();
+    public signal void shown ();
+    public signal void about_to_show_content ();
+
+    PanelExtraHTML view;
+
+    public PanelExtraBox () {
+
+        // We need to register which events we are interested in
+        add_events(Gdk.EventMask.ENTER_NOTIFY_MASK);
+        add_events(Gdk.EventMask.LEAVE_NOTIFY_MASK);
+
+        // Connecting some events, entered, leave and button pressed
+        enter_notify_event.connect(mouse_entered);
+        leave_notify_event.connect(mouse_leave);
+
+        view = new PanelExtraHTML ();
+        view.show_all ();
+        add (view);
+        set_keep_above (true);
+        set_type_hint (Gdk.WindowTypeHint.POPUP_MENU);
+        set_title ("_manokwari_menu_");
+        set_visual (this.screen.get_rgba_visual ());
+
+        Gdk.RGBA c = Gdk.RGBA();
+        c.red = 0.0;
+        c.blue = 0.0;
+        c.green = 0.0;
+        c.alpha = 0.0;
+        override_background_color(StateFlags.NORMAL, c);
+        set_app_paintable(true);
+
+        PanelScreen.move_window (this, Gdk.Gravity.NORTH_EAST);
+
+        view.start();
+
+        hide ();
+
+        button_press_event.connect((event) => {
+
+        stdout.printf("xxx\n");
+            // Only dismiss if within the area
+            // TODO: multihead
+            if (event.x > get_window().get_width()) {
+                  shown ();
+                return true;
+            }
+            return false;
+        });
+
+        screen_size_changed.connect (() =>  {
+            PanelScreen.move_window (this, Gdk.Gravity.NORTH_EAST);
+            queue_resize ();
+        });
+
+        map_event.connect (() => {
+            shown ();
+            PanelScreen.move_window (this, Gdk.Gravity.NORTH_EAST);
+            get_window ().raise ();
+            Utils.grab (this);
+            view.triggerShowAnimation();
+            return true;
+        });
+
+        key_press_event.connect ((e) => {
+            if (Gdk.keyval_name(e.keyval) == "Escape") {
+                if (view.handleEsc () == false) {
+                    dismiss ();
+                }
+            } else if (Gdk.keyval_name(e.keyval) == "Print") {
+                if (Utils.print_screen () == false) {
+                    stdout.printf ("Unable to take screenshot\n");
+                }
+            }
+            return false;
+        });
+
+    }
+
+    public override void get_preferred_width (out int min, out int max) {
+        min = max = COLUMN_WIDTH;
+    }
+
+    public override void get_preferred_height (out int min, out int max) {
+        min = max = PanelScreen.get_primary_monitor_geometry ().height;
+    }
+
+// Mouse entered/leave control function extra panel
+    public bool mouse_entered(Gdk.EventCrossing e) {
+        inside = true;
+        shown();
+        return false;
+    }
+
+    public bool mouse_leave(Gdk.EventCrossing e) {
+        inside = false;
+        try_hide();
+        return false;
+    }
+
+
+    private void dismiss () {
+        if (visible) {
+            stdout.printf("Extra box no dismissed \n");
+            Utils.ungrab (this);
+            shown ();
+        }
+    }
+
+    bool real_hide () {
+        hide ();
+        return false;
+    }
+
+    public void try_hide () {
+        view.triggerHideAnimation();
+        GLib.Timeout.add (500, real_hide);
+        dismissed ();
+    }
+}
diff -Naur manokwari-1.0.9.orig/src/panel-extra-content.vala manokwari-1.0.9/src/panel-extra-content.vala
--- manokwari-1.0.9.orig/src/panel-extra-content.vala	1970-01-01 07:00:00.000000000 +0700
+++ manokwari-1.0.9/src/panel-extra-content.vala	2016-08-24 16:16:17.611462575 +0700
@@ -0,0 +1,30 @@
+using GMenu;
+using Gtk;
+
+// This class provides a header and containter to put contents
+public class PanelExtraContent : VBox {
+    protected VBox bar;
+
+    public signal void extra_clicked ();
+
+    public PanelExtraContent (string? label) {
+
+        if (label != null) {
+            var l = new Label ("");
+            l.set_markup ("<big>\n" + label + "</big>");
+            l.show ();
+            pack_end (l, false, false, 5);
+        }
+        show_all ();
+    }
+
+    public override void get_preferred_width (out int min, out int max) {
+        min = max = 260;
+    }
+
+    public void insert_separator () {
+        var separator = new Separator (Orientation.HORIZONTAL);
+        separator.show ();
+        pack_end (separator, false, false, 10);
+    }
+}
diff -Naur manokwari-1.0.9.orig/src/panel-extra-html.vala manokwari-1.0.9/src/panel-extra-html.vala
--- manokwari-1.0.9.orig/src/panel-extra-html.vala	1970-01-01 07:00:00.000000000 +0700
+++ manokwari-1.0.9/src/panel-extra-html.vala	2016-08-24 16:15:36.838553120 +0700
@@ -0,0 +1,80 @@
+
+using Gtk;
+using WebKit;
+using JSCore;
+
+public class PanelExtraHTML: WebView {
+    string translate_uri (string old) {
+        var uri = old.replace("http://system", "file://" + Config.SYSTEM_PATH + "/");
+        return uri;
+    }
+
+    string translate_theme (string old) {
+        var uri = "file://%s".printf(Utils.get_icon_path (old.replace("theme://", "")));
+        return uri;
+    }
+
+    public PanelExtraHTML () {
+        set_visual (Gdk.Screen.get_default ().get_rgba_visual ());
+
+        Gdk.RGBA c = Gdk.RGBA();
+        c.red = 0.0;
+        c.blue = 0.0;
+        c.green = 0.0;
+        c.alpha = 0.0;
+        override_background_color(StateFlags.NORMAL, c);
+        set_app_paintable(true);
+
+        set_transparent (true);
+        var settings = new WebSettings();
+        settings.enable_file_access_from_file_uris = true;
+        settings.enable_universal_access_from_file_uris = true;
+
+        if (Environment.get_variable("MANOKWARI_DEBUG") == null) {
+            settings.enable_default_context_menu = false;
+        }
+        set_settings(settings);
+
+        resource_request_starting.connect((frame, resource, request, response) => {
+            if (resource.uri.has_prefix("theme://")) {
+                request.set_uri(translate_theme(resource.uri));
+            } else {
+                var uri = translate_uri (resource.uri);
+                request.set_uri(uri);
+            }
+        });
+
+        window_object_cleared.connect ((frame, context) => {
+            Utils.setup_js_class ((JSCore.GlobalContext) context);
+        });
+    }
+
+    public void start() {
+        load_uri ("http://system/bevel.html");
+    }
+
+    public void triggerShowAnimation () {
+        unowned JSCore.Context context = get_focused_frame ().get_global_context();
+        var s = new String.with_utf8_c_string ("extra.prepareShow()");
+        context.evaluate_script (s, null, null, 0, null);
+    }
+
+    public void triggerHideAnimation () {
+        unowned JSCore.Context context = get_focused_frame ().get_global_context();
+        var s = new String.with_utf8_c_string ("extra.prepareHide()");
+        context.evaluate_script (s, null, null, 0, null);
+    }
+
+    public bool handleEsc() {
+        unowned JSCore.Context context = get_focused_frame ().get_global_context();
+        var s = new String.with_utf8_c_string ("extra.handleEsc()");
+        var r = context.evaluate_script (s, null, null, 0, null);
+        if (r.is_boolean (context)) {
+            return r.to_boolean (context);
+        }
+
+        return false;
+    }
+
+
+}
diff -Naur manokwari-1.0.9.orig/src/panel-item.vala manokwari-1.0.9/src/panel-item.vala
--- manokwari-1.0.9.orig/src/panel-item.vala	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/src/panel-item.vala	2016-08-24 12:26:21.072573635 +0700
@@ -62,7 +62,7 @@
         label = new Label ("");
         image = new Image.from_stock (Stock.MISSING_IMAGE, IconSize.LARGE_TOOLBAR); 
         label.show ();
-        label.set_justify(Justification.LEFT);
+        label.set_justify (Justification.LEFT);
         image.show ();
         box.show ();
         box.pack_start (image, false, false, 10);
diff -Naur manokwari-1.0.9.orig/src/panel-menu-box.vala manokwari-1.0.9/src/panel-menu-box.vala
--- manokwari-1.0.9.orig/src/panel-menu-box.vala	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/src/panel-menu-box.vala	2016-08-24 12:39:02.688605210 +0700
@@ -2,10 +2,12 @@
 
 public class PanelMenuBox : PanelAbstractWindow {
     private const int COLUMN_WIDTH = 265;
+    private const int TOP = 24;
 
     public signal void dismissed ();
     public signal void shown ();
     public signal void about_to_show_content ();
+    
     PanelMenuHTML view;
 
     public PanelMenuBox () {
@@ -25,7 +27,8 @@
         view = new PanelMenuHTML ();
         view.show_all ();
         add (view);
-        set_type_hint (Gdk.WindowTypeHint.DOCK);
+        set_keep_above (true);
+        set_type_hint (Gdk.WindowTypeHint.POPUP_MENU);
         set_title ("_manokwari_menu_");
         set_visual (this.screen.get_rgba_visual ());
 
diff -Naur manokwari-1.0.9.orig/src/panel-window.vala manokwari-1.0.9/src/panel-window.vala
--- manokwari-1.0.9.orig/src/panel-window.vala	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/src/panel-window.vala	2016-08-24 15:46:54.186611847 +0700
@@ -40,7 +40,7 @@
         leave_notify_event.connect (() => {
 
             dismiss ();
-            return true; 
+            return true;
         });
 
         map_event.connect (() => {
@@ -93,7 +93,7 @@
         button_press_event.connect ((event) => {
             pager.show_all ();
             pager_shown ();
-            return false; 
+            return false;
         });
 
     }
@@ -105,7 +105,7 @@
 
     public override void get_preferred_height (out int min, out int max) {
         // TODO
-        min = max = 10; 
+        min = max = 10;
     }
 
     public override void get_preferred_width (out int min, out int max) {
@@ -209,7 +209,7 @@
             else
                 set_tooltip_text ("");
             entered ();
-            return false; 
+            return false;
         });
 
         button_press_event.connect ((event) => {
@@ -223,7 +223,7 @@
                 }
                 sync_window_states ();
             }
-            return false; 
+            return false;
         });
 
         drag_motion.connect (() => {
@@ -243,26 +243,26 @@
             icon = window_info.get_icon ();
         }
     }
- 
+
     public override void get_preferred_height (out int min, out int max) {
         update_icon ();
         // TODO
         if (icon != null) {
             min = max = icon.get_width () + Margin * 2;
         } else {
-            min = max = Margin; 
+            min = max = Margin;
         }
     }
 
     public override void get_preferred_width (out int min, out int max) {
-        get_preferred_height (out min, out max); 
+        get_preferred_height (out min, out max);
     }
 
     public override bool draw (Context cr) {
         StyleContext style = get_style_context ();
         style.set_state (state);
 
-        style.render_background (cr, 0, 0, get_allocated_width (), get_allocated_height ()); 
+        style.render_background (cr, 0, 0, get_allocated_width (), get_allocated_height ());
 
         update_icon ();
 
@@ -286,7 +286,7 @@
         menu.attach_to_widget (this, null);
 
         popup_shown = true;
-        menu.popup (null, null, null, button, event_time);
+        menu.popup (null, null, null, button, event_time); 
     }
 
 
@@ -317,13 +317,13 @@
         set_type_hint (Gdk.WindowTypeHint.DOCK);
 
         set_app_paintable(true);
-        
+
         hide ();
         PanelScreen.move_window (this, Gdk.Gravity.NORTH_WEST);
 
         anim = new AnimatedProperty (this);
         anim.set_property ("offset");
-        
+
         anim.frame.connect (() => {
             queue_draw ();
         });
@@ -334,7 +334,7 @@
     }
 
     public override void get_preferred_height (out int min, out int max) {
-        min = max = 34; 
+        min = max = 34;
     }
 
     public override void get_preferred_width (out int min, out int max) {
@@ -351,32 +351,32 @@
         }
         pango.set_font_description (style.get_font (state));
         pango.set_markup ("<big>%s</big>".printf (text), -1);
-      
-        var h = get_window ().get_height (); 
+
+        var h = get_window ().get_height ();
         var icon_margin = 5;
 
         int text_x, text_y, text_w, text_h;
         pango.get_pixel_size (out text_w, out text_h);
-        text_y = h / 2 - text_h / 2; 
+        text_y = h / 2 - text_h / 2;
 
         if (backward) {
             icon_x = start - (icon_margin * 2 + icon_width + text_w);
             if (dir == TextDirection.LTR) {
                 text_x = start - (icon_margin + text_w);
             } else {
-                text_x = start; 
+                text_x = start;
             }
         } else {
             icon_x = start;
             if (dir == TextDirection.LTR) {
                 text_x = start + icon_width + icon_margin;
             } else {
-                text_x = start; 
+                text_x = start;
             }
         }
 
         var occupied = icon_margin * 3 + text_w + icon_width;
-        style.render_background (cr, icon_x + offset - icon_margin, 0, occupied, get_allocated_height ()); 
+        style.render_background (cr, icon_x + offset - icon_margin, 0, occupied, get_allocated_height ());
         if (icon != null) {
             Gdk.cairo_set_source_pixbuf (cr, icon, icon_x + offset, 0);
         }
@@ -402,7 +402,7 @@
         var state = StateFlags.NORMAL;
 
         stack.clear ();
-        style.render_background (cr, 0, 0, get_allocated_width (), get_allocated_height ()); 
+        style.render_background (cr, 0, 0, get_allocated_width (), get_allocated_height ());
 
         if (active_entry != null)  {
             var start_x = -1;
@@ -533,7 +533,7 @@
               next = true;
           } else {
               next = false;
-          } 
+          }
         }
         if (selected == null) {
             selected = first;
@@ -566,14 +566,15 @@
     PanelHotkey hotkey;
     DBusProperties bus;
     Notify.Notification indicator;
+    private Image extra_logo;     ///////// extra logo //////////
 
     public signal void windows_gone (); // Emitted when all windows have gone, either closed or minimized
     public signal void windows_visible (); // Emitted when there is at least one window visible
     public signal void all_windows_visible (); // Emitted when all normal windows visible
     public signal void dialog_opened (); // Emitted when a dialog is opened
     public signal void menu_clicked (); // Emitted when the menu is clicked
-
     public signal void activated (); // Emitted when the window host is activated (the size is getting bigger)
+    public signal void extra_clicked (); /////////// extra panel /////////////// Emitted when the extra is clicked
 
     public bool no_windows_around () {
         update (false);
@@ -595,6 +596,7 @@
         hotkey = new PanelHotkey();
 
         hotkey.bind("<Alt>Tab");
+        hotkey.bind("<Alt>Menu");
         hotkey.bind("Print");
         hotkey.bind("<Mod4>x");
         hotkey.bind("MonBrightnessUp");
@@ -602,15 +604,27 @@
         hotkey.triggered.connect((key) => {
           handleKey(key);
         });
-        
+
         logo = new Image.from_icon_name("distributor-logo", IconSize.LARGE_TOOLBAR);
         var event_box = new EventBox();
+        event_box.set_tooltip_text (_("Menu applications "));
         event_box.add (logo);
         event_box.show_all ();
 
         logo.set_pixel_size (height);
 
-        entry_map = new HashMap <Wnck.Window, PanelWindowEntry> (); 
+        /////////////////// extra button logo ///////////////
+        extra_logo = new Image.from_icon_name("view-list-symbolic", IconSize.LARGE_TOOLBAR);
+        var extra_box = new EventBox();
+        extra_box.set_tooltip_text (_("Widget applications"));
+        extra_box.add (extra_logo);
+        extra_box.show_all ();
+
+        extra_logo.set_pixel_size (height);
+
+        //////////////////// end extra logo /////////////////
+
+        entry_map = new HashMap <Wnck.Window, PanelWindowEntry> ();
 
         tray = new PanelTray ();
         tray.show ();
@@ -647,7 +661,8 @@
         calendar.update_position (height);
         calendar.hide ();
 
-        outer_box.pack_end (pager_entry, false, false, 1);
+        outer_box.pack_end (extra_box, false, false, 3);      ////////// extra //////////
+        outer_box.pack_end (pager_entry, false, false, 0);
         outer_box.pack_end (clock_event, false, false, 0);
         outer_box.pack_end (tray, false, false, 0);
         outer_box.pack_start (event_box, false, false, 6);
@@ -658,7 +673,7 @@
         show();
         reposition ();
 
-        set_struts(); 
+        set_struts();
 
         screen_size_changed.connect (() => {
             queue_resize ();
@@ -714,15 +729,22 @@
         });
 
         configure_event.connect (() => {
-            set_struts(); 
+            set_struts();
             return false;
         });
 
-        event_box.button_press_event.connect (() => {
+        event_box.button_press_event.connect (() => { 
             menu_clicked ();
             return false;
         });
 
+        ///////// extra panel /////////////
+        extra_box.button_press_event.connect (() => {
+            extra_clicked ();
+            return false;
+        });
+        ////////////// end ////////////////
+
         clock_event.button_release_event.connect(() => {
             if (calendar.visible) {
                 calendar.hide ();
@@ -742,7 +764,7 @@
     }
 
     public override void get_preferred_height (out int min, out int max) {
-        min = max = height; 
+        min = max = height;
     }
 
     public void update (bool emit_change_signals) {
@@ -758,7 +780,7 @@
         var num_total_windows = 0;
         var num_windows = 0;
         foreach (unowned Wnck.Window w in screen.get_windows()) {
-            if (!w.is_skip_tasklist () 
+            if (!w.is_skip_tasklist ()
               && (w.get_name() != "_manokwari_menu_")
               && w.is_on_workspace (workspace)) {
                 var e = entry_map [w];
@@ -813,6 +835,9 @@
       else if (key == "<Mod4>x") {
         menu_clicked();
       }
+      else if (key == "<Alt>Menu") {
+        extra_clicked();
+      }
       else if (key == "Print") {
         Utils.print_screen();
       }
@@ -825,7 +850,7 @@
     }
 
     void handleWindowCycle() {
-        descriptions.cycle();      
+        descriptions.cycle();
     }
 
 
diff -Naur manokwari-1.0.9.orig/system/bevel.html manokwari-1.0.9/system/bevel.html
--- manokwari-1.0.9.orig/system/bevel.html	1970-01-01 07:00:00.000000000 +0700
+++ manokwari-1.0.9/system/bevel.html	2016-08-23 23:27:15.198382852 +0700
@@ -0,0 +1,187 @@
+<html>                                                                  
+<head>                                                                  
+<meta charset="utf-8">
+<meta name="viewport" content="width=device-width, initial-scale=1" /> 
+
+	<script type="text/javascript" src="js/jq.js"></script>
+	<script type="text/javascript" src="js/xml2json.js"></script>
+	<script type="text/javascript" src="js/jquery.min.js"></script>
+
+<!-- this is extra bevel -->
+	<link rel="stylesheet" href="css/bevel.css" />
+	<script type="text/javascript" src="js/time.js"></script>
+	<script type="text/javascript" src="js/moment.min.js"></script>
+	<script type="text/javascript" src="js/bevel.js"></script>
+	<link href="widget/weather/css/style.css" rel="stylesheet"><!-- this is extra weather -->	
+	<link href="widget/weather/css/owfont-regular.min.css" rel="stylesheet">
+</head>
+
+<body>
+<!--div id="base"-->
+
+<!--div id="show-top"></div>
+<div id="show-bottom"></div>
+<div id="hide"></div-->
+
+<div id="bevel">
+	<div id="sosial_media">Bookmarks</div>
+	<div class="sosmed">
+		<a href="#" value="Sosmed1" onclick="Run1();" title="http://blankonlinux.or.id"><img src="img/blankon.svg"></a>
+		<a href="#" value="Sosmed2" onclick="Run2();" title="http://panduan.blankonlinux.or.id"><img src="img/blankon-tutorial.svg"></a>
+		<a href="#" value="Sosmed3" onclick="Run3();" title="http://facebook.com/blankonlinux"><img src="img/facebook.svg"></a>
+		<a href="#" value="Sosmed4" onclick="Run4();" title="http://twitter.com/blankonlinux"><img src="img/twitter.svg"></a>
+	</div>
+	<div id="player">
+		<div class="header">Music Player</div>
+		<div class="control">
+		<div id="buttons">
+			<li id="stop" href="#" value="" onclick="Stop();" title="Stop/Quit"><img src="img/player/stop.svg"></li>
+			<li id="prev" href="#" value="" onclick="Prev();" title="Previous"><img src="img/player/prev.svg"></li>
+			<li id="play" href="#" value="" onclick="Play();" title="Play"><img src="img/player/play.svg"></li>
+			<li id="pause" href="#" value="" onclick="Pause();" title="Pause"><img src="img/player/pause.svg"></li>
+			<li id="next" href="#" value="" onclick="Next();" title="Next"><img src="img/player/next.svg"></li>
+			<li id="repeat" href="#" value="" onclick="Repeat();" title="Repeat"><img src="img/player/repeat.svg"></li>
+		 </div>
+		<!--div id="volume-section">
+			<div id="volume-wrapper">
+				<img id="volume" src="img/volume.png"><input id="volumeslider" type="range" min="0" max="100" value="100">
+			</div>
+		</div-->
+
+	<script src="js/jquery.js"></script>
+	<script src="js/main.js"></script>
+	</div>
+	
+<!-- Widget Weather -->
+
+<div id="container">
+	<div id="tittle"><span>Weather</span></div>
+	<div class="border">
+		<div class="settings"><img src="widget/weather/img/settings.svg"></div>
+		<div class="sync"><img src="widget/weather/img/sync.svg"></div>
+	</div>
+	<div id="actualWeather">
+		<div id="city"><span></span></div>
+		<div id="ow-code"><i id="icon"></i></div>
+		<div class="details">
+			<div class="left" id="temperature"></div>
+			<div class="right">
+					<span id="windSpeed"></span><span id="windUnit"></span><br>
+					<span id="humidity"></span>
+			</div>
+		</div>
+		<div class="week">
+			<div id="0">
+				<span class="day"></span>
+				<span class="code"></span>
+				<span class="temp"></span>
+			</div>
+			<div id="1">
+				<span class="day"></span>
+				<span class="code"></span>
+				<span class="temp"></span>
+			</div>
+			<div id="2">
+				<span class="day"></span>
+				<span class="code"></span>
+				<span class="temp"></span>
+			</div>
+			<div id="3">
+				<span class="day"></span>
+				<span class="code"></span>
+				<span class="temp"></span>
+			</div>
+			<div id="4">
+				<span class="day"></span>
+				<span class="code"></span>
+				<span class="temp"></span>
+			</div>
+		</div>
+		</div>
+	</div>
+	<div id="locationModal">
+		<div class="credits">
+		<img src="widget/weather/img/back.svg">
+		<h1 style="text-align: center">Tekukur 1.0</h1>
+		<p>Tekukur simple weather BlankOn Linux</p>
+
+		<h3 style="text-align: center">Thanks for using.</h3><br><br>
+	</div>
+	<div class="input">
+	<input type="text" placeholder="Location"><span class="loader"></span><br>
+	e.g Bandung, Jakarta<br>
+	</div>
+
+	<div class="full">
+		<div class="toggleswitch measurement">
+			<span data-type="c">R</span><span data-type="f">T</span><span data-type="k">K</span>
+		</div>
+		<div class="toggleswitch speed">
+			<span data-type="mph">mph</span><span data-type="kph">kph</span><span data-type="ms">m/s</span>
+		</div>
+		<div class="color">
+			<span data-color="101010"></span>
+			<span data-color="008000"></span>
+			<span data-color="004F1B"></span>
+			<span data-color="C9A100"></span>
+			<span data-color="C83608"></span>
+			<span data-color="920200"></span>
+			<span data-color="A52A2A"></span>
+			<span data-color="004E99"></span>
+			<span data-color="8A02A9"></span>
+			<span data-color="111837"></span>
+			<span data-color="gradient"></span>
+		</div>
+		<br>
+		<!--a class="btn" tag="credits">About</a-->
+		<br><br><br><br>
+	</div>
+	</div>
+	<div id="errorMessage">
+		Could not connect to server.<br>
+		Check your connection.<br>
+		<button class="btn">Try Again</button>
+	</div>
+	<script src="widget/weather/js/jquery.min.js"></script>
+	<script src="widget/weather/js/script.js"></script>
+
+<!-- System System --->
+	<div id="system_settings">System Settings</div>
+	<div class="system">
+		<li href="#" value="" onclick="RunWallpaper()"><img src="img/icons/personalization_normal.png"><span>Background</span></li>
+		<li href="#" value="" onclick="RunAccount()"><img src="img/icons/account_normal.png"><span>User</span></li>
+		<li href="#" value="" onclick="RunDate()"><img src="img/icons/date_time_normal.png"><span>Date/Time</span></li>
+		<li href="#" value="" onclick="RunOnline()"><img src="img/icons/default_applications_normal.png"><span>Online</span></li>
+		<li href="#" value="" onclick="RunRegional()"><img src="img/icons/network_normal.png"><span>Regional</span></li>
+		<li href="#" value="" onclick="RunDisplay()"><img src="img/icons/display_normal.png"><span>Display</span></li>
+		<li href="#" value="" onclick="RunNetwork()"><img src="img/icons/wireless.png"><span>Network</span></li>
+		<li href="#" value="" onclick="RunKeyboard()"><img src="img/icons/keyboard_normal.png"><span>Keyboard</span></li>
+		<li href="#" value="" onclick="RunMouse()"><img src="img/icons/mouse_normal.png"><span>Mouse</span></li>
+		<li href="#" value="" onclick="RunSound()"><img src="img/icons/sound_normal.png"><span>Sound</span></li>
+		<li href="#" value="" onclick="RunBluetooth()"><img src="img/icons/bluetooth_normal.png"><span>Bluetooth</span></li>
+		<li href="#" value="" onclick="RunPower()"><img src="img/icons/power_normal.png"><span>Power</span></li>
+		<li href="#" value="" onclick="RunInfo()"><img src="img/icons/system_info_normal.png"><span>Detail</span></li>
+		<li href="#" value="" onclick="RunPrinter()"><img src="img/icons/printer.png"><span>Printer</span></li>
+		<li href="#" value="" onclick="RunShare()"><img src="img/icons/share.png"><span>Sharing</span></li>
+	</div>
+	
+	<div id="time" class="clock"><span id="clock-time">00:00</span><span id="clock-date">16 Juli 2013</span>
+	</div>
+	</div>
+<!--script>
+$(document).ready(function(){
+    $("#show-top, #show-bottom").hover(function(){
+        $("#bevel").css({
+          right: '0px'
+		});
+    $("#hide").hover(function(){
+        $("#bevel").css({
+          right: '-267px'
+		});
+		});
+	});
+});
+</script-->
+</body>
+
+</html>
diff -Naur manokwari-1.0.9.orig/system/css/bevel.css manokwari-1.0.9/system/css/bevel.css
--- manokwari-1.0.9.orig/system/css/bevel.css	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/system/css/bevel.css	2016-08-24 12:08:57.960530390 +0700
@@ -24,33 +24,33 @@
 	position: absolute;
 	width: 100%;
 	height: 100%;
-	background: transparent;
+	background-color: transparent;
 	right: 0px;
 	top: 0px;
 	z-index: -100;
 }
 
 #bevel {
-	background: #111111;
+	background-color: rgba(17,17,17,0.98);
 	color: #aaa;
-	height: 104%;
+	height: 100%;
 	width: 265px;
 	overflow-x: hidden;
 	overflow-y: auto;
-	right: -266px;
+	right: 0px;
 	top: 0px;
-	bottom:0px;
+	bottom: 0px;
 	position:absolute;
 	z-index: 150;
-	-webkit-transition: right 250ms ease-in-out;
 }
-/*#bevel:hover {
-	right: 0px;
-}*/
+
+.ui-animation-slide {
+    -webkit-transition: all 0.2s ease-in-out;
+}
 
 /* Sosmed */
 #sosial_media {
-	margin-top: 30px;
+	margin-top: 5px;
 	padding: 2px 0;
 	width:100%;
 	height: 24px;
@@ -85,8 +85,8 @@
 
 /* Music Player */
 #player, #control {
-	margin-top: 10px;
-	padding-top: 8px;
+	margin-top: 15px;
+	padding-top: 5px;
 	color: #aaa;
 	width:100%;
 	height: auto;
@@ -97,7 +97,7 @@
 	cursor:default;
 }
 #buttons{
-    margin-top:20px;
+    margin-top:15px;
 	width:auto;
 	height: 34px;
 	float:center;
@@ -105,10 +105,10 @@
 #buttons #play img,
 #buttons #pause img {
 	position:absolute;
-	width: 32px;
-	height: 32px;
-	margin-top:-31px;
-	margin-left:-22px;
+	width: 24px;
+	height: 24px;
+	margin-top:-28px;
+	margin-left:-19px;
 	padding: 5px;
 	float:center;
 	background: #333;
@@ -160,10 +160,79 @@
 #buttons .selected {
 	background-color: #444;
 	-webkit-transition: all 0.2s ease-in-out;
-	-webkit-box-shadow: inset 0px 0px 1px 15px #492F2F;  /*#882626;*/
+	-webkit-box-shadow: inset 0px 0px 1px 15px #4084D6;  /*#882626;*/
 	border: 2px solid #000;
 }
 
+#volume-section{
+	width: 100%;
+	height: 34px;
+	float: right;
+	display: table;
+}
+
+#volume-wrapper{
+	text-align: center;
+	display: table-cell;
+  	vertical-align: middle;
+}
+#volume{
+	width: 20px;
+	height: 20px;
+	border: none;
+	vertical-align: 1px;
+	padding-right: 10px;
+}
+#volumeslider{
+	-webkit-appearance: none;
+    background:#ccc;
+    width: 70%;
+    height:1px;
+    vertical-align: 10px;
+}
+
+#volumeslider::-webkit-slider-thumb{
+	-webkit-appearance: none;
+	background:url(../img/player/knob.png) no-repeat;
+	height: 12px;
+	width: 12px;
+}
+div#progress-section{ 
+	position: absolute;
+   	width: 40%;
+   	left: 30%;
+   	height: 100%;
+   	display: table;
+}  
+
+div#progressbar-wrapper{
+	text-align: center;
+	height: 100%;
+	font-size: 14px;
+	display: table-cell;
+  	vertical-align: middle;
+}
+
+.progress{
+	display: inline-block;
+}
+
+#current-song-info{
+	width: 38%;
+	height: 35px;
+	margin: 0 auto;
+	word-wrap: break-word;
+	vertical-align: text-bottom;
+}
+
+div#progressbar{
+	background: url(../img/progressbar.png) no-repeat;
+	margin-left: 8px;
+	margin-right: 8px;
+	width: 48%;
+	height: 10px;
+}
+
 /* System setting */
 #system_settings {
 	margin-top: 0px;
@@ -224,7 +293,7 @@
 	background: #1a1a1a;
 	color: #bbb;
 	bottom: 0px;
-	padding: 5px 0 10px;
+	padding: 5px 0 34px;
     cursor: default;
     text-align: center;
     border-top: 1px solid #0C0C0C;
@@ -235,7 +304,7 @@
 	font-family: 'TrendexLightSSi.ttf';
 	display: block;
 	font-size: 42px;
-	letter-spacing: 0;
+	letter-spacing: 2;
 	text-shadow: 0 0px 2px #000;
 }
 #clock-date {
@@ -291,3 +360,20 @@
     font-weight: 400;
     src: local('Ubuntu Condensed'), local('UbuntuCondensed-Regular'), url('../fonts/ubuntu.woff') format('woff');
 }
+
+body {
+	background-color: transparent;
+	overflow: hidden;
+	-webkit-user-select: none;
+	padding: 0px;
+	margin: 0px;
+	font-family: Cantarell, Sans-Serif;
+}
+
+#base {
+	background-color: transparent;
+	position: absolute;
+	z-index: 3;
+	width: 100%;
+	height: 100%;
+}
diff -Naur manokwari-1.0.9.orig/system/css/style.css manokwari-1.0.9/system/css/style.css
--- manokwari-1.0.9.orig/system/css/style.css	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/system/css/style.css	2016-08-24 12:09:02.996530599 +0700
@@ -211,10 +211,10 @@
     position: fixed;
     clear: both;
     bottom: 0;
-    height: 65px;
+    height: 90px;
     width: 250px; 
     margin: 0px 0px -65px 0px;
-    padding: 10px 10px 0px 15px;
+    padding: 10px 10px 0px 20px;
     z-index: 100;
     background: #1a1a1a; 
     overflow: hidden;
@@ -235,6 +235,7 @@
     width: 48;
     height: 48;
     margin-right: 10px;
+	margin-top: 3px;
     cursor: pointer;
 	opacity: 0.8;
 	-webkit-transition: all 0.2s ease-out;
@@ -250,6 +251,7 @@
     margin: 0px;
     padding-top:5px;
     font-family: 'Aurulent Sans';
+	opacity: 0.8;
 }
 
 #userAccount span {
@@ -257,6 +259,7 @@
     color: #e2e2e2;
     text-shadow: 0 1px 1px #000;
     font-family: 'Aurulent Sans';
+    opacity: 0.8;
 }
 
 ::-webkit-scrollbar {
diff -Naur manokwari-1.0.9.orig/system/desktop.html manokwari-1.0.9/system/desktop.html
--- manokwari-1.0.9.orig/system/desktop.html	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/system/desktop.html	2016-08-17 03:58:24.031382543 +0700
@@ -4,21 +4,12 @@
 
 <meta name="viewport" content="width=device-width, initial-scale=1" /> 
 	<link rel="stylesheet" href="css/desktop.css" />
-	<link rel="stylesheet" href="css/bevel.css" />
 
 	<script type="text/javascript" src="js/jq.js"></script>
 	<script type="text/javascript" src="js/xml2json.js"></script>
 	<script type="text/javascript" src="js/xml-background.js"></script>
 	<script type="text/javascript" src="js/desktop.js"></script>
 	
-<!-- this is extra bevel -->
-	<script type="text/javascript" src="js/time.js"></script>
-	<script type="text/javascript" src="js/moment.min.js"></script>
-	<script type="text/javascript" src="js/bevel.js"></script>
-<!-- this is extra weather -->	
-	<link href="widget/weather/css/style.css" rel="stylesheet">
-	<link href="widget/weather/css/owfont-regular.min.css" rel="stylesheet">
-	
 </head>
 
 <body>
@@ -28,165 +19,7 @@
     <!-- h1 id="debug"></h1 -->
 	<div id="launcher"></div>
 
-<div id="show-top"></div>
-<div id="show-bottom"></div>
-
-<div id="hide"></div>
-<div id="bevel">
-	<div id="sosial_media">Bookmarks</div>
-	<div class="sosmed">
-		<a href="#" value="Sosmed1" onclick="Run1();" title="http://blankonlinux.or.id"><img src="img/blankon.svg"></a>
-		<a href="#" value="Sosmed2" onclick="Run2();" title="http://panduan.blankonlinux.or.id"><img src="img/blankon-tutorial.svg"></a>
-		<a href="#" value="Sosmed3" onclick="Run3();" title="http://facebook.com/blankonlinux"><img src="img/facebook.svg"></a>
-		<a href="#" value="Sosmed4" onclick="Run4();" title="http://twitter.com/blankonlinux"><img src="img/twitter.svg"></a>
-	</div>
-	<div id="player">
-		<div class="header">Music Player</div>
-		<div class="control">
-		<div id="buttons">
-			<li id="stop" href="#" value="" onclick="Stop();" title="Stop/Quit"><img src="img/player/stop.svg"></li>
-			<li id="prev" href="#" value="" onclick="Prev();" title="Previous"><img src="img/player/prev.svg"></li>
-			<li id="play" href="#" value="" onclick="Play();" title="Play"><img src="img/player/play.svg"></li>
-			<li id="pause" href="#" value="" onclick="Pause();" title="Pause"><img src="img/player/pause.svg"></li>
-			<li id="next" href="#" value="" onclick="Next();" title="Next"><img src="img/player/next.svg"></li>
-			<li id="repeat" href="#" value="" onclick="Repeat();" title="Repeat"><img src="img/player/repeat.svg"></li>
-		 </div>
-
-	<!--script src="js/jquery.js"></script-->
-	<script src="js/main.js"></script>
-	</div>
-	
-<!-- Widget Weather -->
 
-<div id="container">
-	<div id="tittle"><span>Weather</span></div>
-	<div class="border">
-		<div class="settings"><img src="widget/weather/img/settings.svg"></div>
-		<div class="sync"><img src="widget/weather/img/sync.svg"></div>
-	</div>
-	<div id="actualWeather">
-		<div id="city"><span></span></div>
-		<div id="ow-code"><i id="icon"></i></div>
-		<div class="details">
-			<div class="left" id="temperature"></div>
-			<div class="right">
-					<span id="windSpeed"></span><span id="windUnit"></span><br>
-					<span id="humidity"></span>
-			</div>
-		</div>
-		<div class="week">
-			<div id="0">
-				<span class="day"></span>
-				<span class="code"></span>
-				<span class="temp"></span>
-			</div>
-			<div id="1">
-				<span class="day"></span>
-				<span class="code"></span>
-				<span class="temp"></span>
-			</div>
-			<div id="2">
-				<span class="day"></span>
-				<span class="code"></span>
-				<span class="temp"></span>
-			</div>
-			<div id="3">
-				<span class="day"></span>
-				<span class="code"></span>
-				<span class="temp"></span>
-			</div>
-			<div id="4">
-				<span class="day"></span>
-				<span class="code"></span>
-				<span class="temp"></span>
-			</div>
-		</div>
-		</div>
-	</div>
-	<div id="locationModal">
-		<div class="credits">
-		<img src="widget/weather/img/back.svg">
-		<h1 style="text-align: center">Tekukur 1.0</h1>
-		<p>Tekukur simple weather BlankOn Linux</p>
-
-		<h3 style="text-align: center">Thanks for using.</h3><br><br>
-	</div>
-	<div class="input">
-	<input type="text" placeholder="Location"><span class="loader"></span><br>
-	e.g Bandung, Jakarta<br>
-	</div>
-
-	<div class="full">
-		<div class="toggleswitch measurement">
-			<span data-type="c">R</span><span data-type="f">T</span><span data-type="k">K</span>
-		</div>
-		<div class="toggleswitch speed">
-			<span data-type="mph">mph</span><span data-type="kph">kph</span><span data-type="ms">m/s</span>
-		</div>
-		<div class="color">
-			<span data-color="101010"></span>
-			<span data-color="008000"></span>
-			<span data-color="004F1B"></span>
-			<span data-color="C9A100"></span>
-			<span data-color="C83608"></span>
-			<span data-color="920200"></span>
-			<span data-color="A52A2A"></span>
-			<span data-color="004E99"></span>
-			<span data-color="8A02A9"></span>
-			<span data-color="111837"></span>
-			<span data-color="gradient"></span>
-		</div>
-		<br>
-		<!--a class="btn" tag="credits">About</a-->
-		<br><br><br><br>
-	</div>
-	</div>
-	<div id="errorMessage">
-		Could not connect to server.<br>
-		Check your connection.<br>
-		<button class="btn">Try Again</button>
-	</div>
-	<script src="widget/weather/js/jquery.min.js"></script>
-	<script src="widget/weather/js/script.js"></script>
-
-<!-- System System --->
-	<div id="system_settings">System Settings</div>
-	<div class="system">
-		<li href="#" value="" onclick="RunWallpaper()"><img src="img/icons/personalization_normal.png"><span>Background</span></li>
-		<li href="#" value="" onclick="RunAccount()"><img src="img/icons/account_normal.png"><span>User</span></li>
-		<li href="#" value="" onclick="RunDate()"><img src="img/icons/date_time_normal.png"><span>Date/Time</span></li>
-		<li href="#" value="" onclick="RunOnline()"><img src="img/icons/default_applications_normal.png"><span>Online</span></li>
-		<li href="#" value="" onclick="RunRegional()"><img src="img/icons/network_normal.png"><span>Regional</span></li>
-		<li href="#" value="" onclick="RunDisplay()"><img src="img/icons/display_normal.png"><span>Display</span></li>
-		<li href="#" value="" onclick="RunNetwork()"><img src="img/icons/wireless.png"><span>Network</span></li>
-		<li href="#" value="" onclick="RunKeyboard()"><img src="img/icons/keyboard_normal.png"><span>Keyboard</span></li>
-		<li href="#" value="" onclick="RunMouse()"><img src="img/icons/mouse_normal.png"><span>Mouse</span></li>
-		<li href="#" value="" onclick="RunSound()"><img src="img/icons/sound_normal.png"><span>Sound</span></li>
-		<li href="#" value="" onclick="RunBluetooth()"><img src="img/icons/bluetooth_normal.png"><span>Bluetooth</span></li>
-		<li href="#" value="" onclick="RunPower()"><img src="img/icons/power_normal.png"><span>Power</span></li>
-		<li href="#" value="" onclick="RunInfo()"><img src="img/icons/system_info_normal.png"><span>Detail</span></li>
-		<li href="#" value="" onclick="RunPrinter()"><img src="img/icons/printer.png"><span>Printer</span></li>
-		<li href="#" value="" onclick="RunShare()"><img src="img/icons/share.png"><span>Sharing</span></li>
-	</div>
-	
-	<div id="time" class="clock"><span id="clock-time">00:00</span><span id="clock-date">16 Juli 2013</span>
-	</div>
-</div>
-
-<!-- this is function bevel hover -->
-<script> 
-$(document).ready(function(){
-    $("#show-top, #show-bottom").hover(function(){
-        $("#bevel").css({
-          right: '0px'
-		});
-    $("#hide").hover(function(){
-        $("#bevel").css({
-          right: '-267px'
-		});
-		});
-	});
-});
 </script>
 
 <!-- this is script disble tab key -->
diff -Naur manokwari-1.0.9.orig/system/js/bevel.js manokwari-1.0.9/system/js/bevel.js
--- manokwari-1.0.9.orig/system/js/bevel.js	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/system/js/bevel.js	2016-08-22 23:20:54.785494927 +0700
@@ -11,24 +11,6 @@
 	Utils.run_command("xdg-open https://twitter.com/BlankOnLinux/");
 }
 
-//mocp
-/*function MocpStart(){Utils.run_command("audacious");}
-
-function MocpState(){
-	document.getElementById("MocpTitle2").innerHTML = Utils.run_command('mocp -Q %title');
-}*/
-
-/*function MocpBackward(){Utils.run_command("audacious -r");}
-function MocpPause(){Utils.run_command("audacious -u");}
-function MocpPlay(){
-	if(Utils.run_command("audacious") == false){
-		Utils.run_command("audacious");
-		Utils.run_command("audacious -p");
-	}
-	else{
-		Utils.run_command("audacious -p");
-	}
-}*/
 function Play(){Utils.run_command("audacious -p");}
 function Stop(){Utils.run_command("audtool shutdown");}
 function Prev(){Utils.run_command("audacious -r");}
@@ -43,15 +25,8 @@
 		else{$('#repeat').addClass("selected");}
 	});
 
-	//$('#shuffle').click(function(){
-		//if($('#shuffle').hasClass("selected")){ $('#shuffle').removeClass("selected");}
-		//else{$('#shuffle').addClass("selected");}
-	//});
-
 });
 
-// gnome control center
-
 function RunWallpaper() {
 	Utils.run_command("gnome-control-center background");
 }
@@ -97,3 +72,93 @@
 function RunShare() {
 	Utils.run_command("gnome-control-center sharing");
 }
+
+//  slide  //
+
+var extra = (function() {
+    var activePage = null;
+    var prepareShow = function() {
+        $("#bevel").css("right", "0px");
+    }
+
+    var prepareHide = function() {
+        changePage($("#bevel"), {
+            transition: "none"
+        });
+        $("#bevel").addClass("ui-animation-slide");
+        $("#bevel").css("right", "-" + window.outerWidth + "px");
+    }
+
+    var changePage = function(page) {
+        if (typeof page === "undefined" || page.length == 0) {
+            console.log("The specified page is invalid");
+            return;
+        }
+
+        var withAnimation = true;
+
+        if (arguments.length == 2) {
+            withAnimation = (arguments[1].transition != "none");
+        }
+        var width = window.outerWidth;
+
+        if (activePage == null) {
+            activePage = $("#bevel");
+            activePage.addClass("ui-animation-slide");
+        }
+
+        if (activePage == page) {
+            console.log("Trying to change to same page, exiting.");
+            return;
+        }
+
+        page.show();
+        if (withAnimation == false) {
+            activePage.removeClass("ui-animation-slide");
+        }
+        page.removeClass("ui-animation-slide");
+
+        if (page.attr("id") == "bevel") {
+            console.log("going to bevel");
+
+            if (withAnimation) {
+                page.css("right", "-" + width + "px");
+            }
+            activePage.css("right", width + "px");
+        } else {
+            console.log("going away from bevel");
+
+            if (withAnimation) {
+                page.css("right", width + "px");
+            }
+            page.css("right");
+            activePage.css("right", "-" + width + "px");
+        }
+
+        if (withAnimation) {
+            page.addClass("ui-animation-slide");
+        }
+        page.css("right", "0px");
+        activePage = page;
+    }
+
+    var handleEsc = function(e) {
+        if (activePage != null) {
+            if (activePage.attr("id") != "bevel") {
+                changePage($("#bevel"));
+                return true;
+            }
+        }
+        return false; // Not handled
+    }
+
+    return {
+        handleEsc: handleEsc,
+        prepareHide: prepareHide,
+        prepareShow: prepareShow,
+    }
+})();
+
+//$(document).ready(function() {
+//    extra.init ();
+//});
diff -Naur manokwari-1.0.9.orig/system/widget/weather/config.xml manokwari-1.0.9/system/widget/weather/config.xml
--- manokwari-1.0.9.orig/system/widget/weather/config.xml	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/system/widget/weather/config.xml	1970-01-01 07:00:00.000000000 +0700
@@ -1,32 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<!--widget xmlns="http://www.w3.org/ns/widgets" id="https://github.com/mijortsa/Cuaca/" version="1.0" viewmodes="fullscreen">
-    <icon src="icon_128.png"/>
-    <content src="*.html"/>
-<name>Cuaca</name>
-</widget-->
-
-<widget xmlns       = "http://www.w3.org/ns/widgets"
-        id          = ""
-        version     = "git"
-        height      = "768"
-        width       = "1024">
- 
-  <name short="cuaca">
-    Cuaca
-  </name>
-
-  <description>
-    BlankOn Cuaca
-    
-  </description>
-
-  <author href=""></author>
-
-  <icon src="icon_48.png"/>
-  <content src="index.html"/>
-
-  <license>
-    (cc by sa version 2.0) 
-  </license>
-</widget>
-
diff -Naur manokwari-1.0.9.orig/system/widget/weather/css/style.css manokwari-1.0.9/system/widget/weather/css/style.css
--- manokwari-1.0.9.orig/system/widget/weather/css/style.css	2016-07-31 20:10:39.000000000 +0700
+++ manokwari-1.0.9/system/widget/weather/css/style.css	2016-08-19 10:23:02.023504060 +0700
@@ -11,7 +11,7 @@
     overflow-y:hidden;
     overflow-x:hidden;
     border-top: 1px solid #0a0a0a;
-    -webkit-transition: 300ms background ease;
+    -webkit-transition: 200ms background ease;
     -webkit-transform-origin: 0 0;
     opacity: 0.8;
 }
@@ -22,8 +22,8 @@
 	text-align: left;
 	color: #eee;
 	font-size: 14px;
-	padding: 8px;
-	margin-left: 4px;
+	padding-top: 8px;
+	margin-left: 15px;
 }
 .full {
 	margin-top: 5px;
@@ -185,7 +185,7 @@
     width: 100%;
     height: 141px;
     bottom: 0px;
-    top: 215px;
+    top: 190px;
     padding-top: 2px; 
     padding-left: 0px;
     padding-bottom: 0px;
@@ -197,7 +197,7 @@
     position: absolute;
     width: 100%;
     height: 120px;
-    top: 215px;
+    top: 190px;
     padding-top: 20px;
     -webkit-transform-origin: 0 0;
 }
@@ -280,7 +280,7 @@
     line-height: 30px;
     height: 28px;
     padding: 0 2px 0 5px;
-    margin-top: 4px;
+    margin: 2px 0px 0px 2px;
 }
 #locationModal .loader.loading {
     font-family:"ClimaconsRegular";
